src/routers/alpha-router/functions/get-candidate-pools.ts

```diff
@@ -365,6 +380,450 @@ function findCrossProtocolMissingPools<
   return selectedPools;
 }
 
+export type V4CandidatePools = {
+  poolAccessor: V4PoolAccessor;
+  candidatePools: CandidatePoolsBySelectionCriteria;
+  subgraphPools: V4SubgraphPool[];
+};
+
+export async function getV4CandidatePools({
```

Comment:
I see very few differences between v4 and v3 (tick sizing, naming + one more point in the middle). Shall we try to collapse the functions?

Reply:
I'm hesitant to refactor in this PR, due to the large function. I'd be more confident to refactor V3CandidatePools against V4CandidatePools when we add more testing coverages against V3CandidatePools. Also, I don't see we will have large amount of v4 subgraph pools for a while, once we launch the v4 routing, which leaves us some time to address this tech debt.

I'm thinking to add a tech debt ticket for this. lmk what you think.

Reply:
can yoou add the ticket? then you can resolve the comment:)

Reply:
Done! https://linear.app/uniswap/issue/ROUTE-241/tech-debt-refactor-getv3candidatepools-against-getv4candidatepools